portalnet/src/overlay_service.rs

```diff
@@ -1432,25 +1447,30 @@ where
 
             // send the content to the acceptor over a uTP stream
             if let Err(err) = stream.write(&content_payload).await {
+                metrics.report_outbound_utp_tx(&protocol, false);
                 warn!(
                     %err,
                     cid.send,
                     cid.recv,
                     peer = ?cid.peer.client(),
                     "Error sending content over uTP connection"
                 );
+                return;
             }
 
             // close uTP connection
             if let Err(err) = stream.shutdown() {
+                metrics.report_outbound_utp_tx(&protocol, false);
```

Comment:
I'm not sure if we want to report this as uTP stream failure, I'm not very familiar with Jacob's uTP design but I think this shutdown is to make the stream inactive and disconnected, we have already sent/received all packets?

Reply:
Hm, yeah I think this looks like a failure, but I agree it's not 100%. AFAICT, there's no current API to be totally sure. That's discussed a little bit here: https://github.com/ethereum/utp/issues/89